### ISSUE TITLE
Enhancement: <Please upgrade typescript-eslint to support Typescript 4.8>

### ISSUE THREAD

**Issue by lainbo** [typescript-eslint/typescript-eslint#5590]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

In TypeScript versions greater than 4.8, using "eslint . --fix", there will be 
```
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.8.0

YOUR TYPESCRIPT VERSION: 4.8.2

Please only submit bug reports when using the officially supported version.
```
 [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint) in 5.36.0 added support for typescript4.8, please upgrade the relevant dependencies

### Fail

```typescript
var replace = 'me';
```

### Pass

```typescript
const replace = 'me';
```

**Comment by bradzacher** [comment]
We have full support for TS 4.8

You likely have an old version of our tooling installed.

https://typescript-eslint.io/docs/linting/troubleshooting/#how-do-i-check-to-see-what-versions-are-installed

> typescript-eslint in 5.36.0 added support for typescript4.8, please upgrade the relevant dependencies

This is typescript-eslint......

**Comment by lainbo** [comment]
> We have full support for TS 4.8
> 
> You likely have an old version of our tooling installed.
> 
> https://typescript-eslint.io/docs/linting/troubleshooting/#how-do-i-check-to-see-what-versions-are-installed
> 
> > typescript-eslint in 5.36.0 added support for typescript4.8, please upgrade the relevant dependencies
> 
> This is typescript-eslint......

I'm sorry I posted in the wrong repository
